hack/monitor_prow.py

```diff
@@ -42,28 +49,94 @@ def highlight(log_dir, dc):
                 '--selector', 'component={}'.format(dc),
                 '--output', 'jsonpath={.items[*].metadata.name}').split():
+            debug("deployment/{}: pod/{}: gathering info".format(dc, pod))
+            lines.extend(renderFlavor(pod, dc))
             cmd = [
                 'oc', 'logs', '--namespace', 'ci', '--since', '20m',
                 'pod/{}'.format(pod)]
             if pod.startswith('deck-internal'):
                 cmd += ['--container', 'deck']
+            debug("deployment/{}: pod/{}: getting logs".format(dc, pod))
             for l in exec_cmd(*cmd).splitlines():
                 if warn in l:
-                    lines.append(YELLOW + l + RESET)
+                    log_lines.append(YELLOW + l + RESET)
                 elif error in l or fatal in l:
-                    lines.append(RED + l + RESET)
+                    log_lines.append(RED + l + RESET)
+
+        if not log_lines and not lines:
+            header = header + " " + GREEN + "OK" + RESET
         with open(log, 'w') as f:
-            if lines:
-                f.write('\n'.join([header, *lines[-3:]]))
+            f.write('\n'.join([header, *lines, *log_lines[-5:]]))
         time.sleep(60)
 
 
+def renderHeader(dc):
+    debug("deployment/{}: rendering header".format(dc))
+    rawdc = json.loads(exec_cmd(
+        'oc', 'get', 'deployment/{}'.format(dc),
+        '--namespace', 'ci', '--output', 'json'))
+    spec = rawdc.get("spec", {})
+    status = rawdc.get("status", {})
+    desired = spec.get("replicas", 0)
+    current = status.get("replicas", 0)
+    updated = status.get("updatedReplicas", 0)
+    available = status.get("availableReplicas", 0)
+    version = "<unknown-version>"
+    containers = spec.get("template", {}).get("spec", {}).get("containers", [])
+    for container in containers:
+        if container.get("name", "") == dc:
+            image = container.get("image", "")
+            version = image[image.index(":")+1:]
+    headerColor = ''
+    if desired != current:
+        headerColor = RED
+
+    message = '{} at {} [{}/{}]'.format(dc, version, current, desired)
+    if updated != desired:
+        message += ' ({} stale replicas)'.format(desired - updated)
+    if available != desired:
+        message += ' ({} unavailable replicas)'.format(desired - available)
+    header = '{}{}{}:{}'.format(BOLD, headerColor, message, RESET)
+    debug("deployment/{}: got header {}".format(dc, header))
+    return header
+
+
+def renderFlavor(pod, dc):
+    debug("deployment/{}: pod/{}: rendering flavor".format(dc, pod))
+    lines = []
+    raw = json.loads(exec_cmd('oc', 'get', 'pod/{}'.format(pod),
+                              '--namespace', 'ci', '--output', 'json'))
+    for container in raw.get("status", {}).get("containerStatuses", []):
+        debug("pod/{}: handling status for container {}".format(pod, container.get("name", "")))
+        if container.get("name", "") == dc:
```

Comment:
The `default=None` thing again, plus in the rest of this method. Especially using `""` as the not-present indicator is suspicious, this should be either `None`, or the matching-type empty value of what `get` is expected to return (usually dictionaries here).